test/jdk/sun/security/krb5/Krb5NameEquals.java

```diff
@@ -22,57 +22,133 @@
+
+        var isNative = false;
+
+        if (os.equals("Linux") || os.equals("Mac OS X")) {
+
+            isNative = true;
+
+            // Not all *nix has native GSS libs installed
+            final var krb5ConfigCheck = ProcessTools.executeCommand(
+                    "sh",
+                    "-c",
+                    "krb5-config --libs 2> /dev/null"
+            );
+
+            if (krb5ConfigCheck.getExitValue() != 0) {
 
-    static {
-        Oid temp = null;
-        try {
-            temp = new Oid("1.2.840.113554.1.2.2"); // KRB5
-        } catch (Exception e) {
-            // should never happen
+                // Fedora has a different path
+                final var krb5ConfigCheck2 = ProcessTools.executeCommand(
+                        "sh",
+                        "-c",
+                        "/usr/kerberos/bin/krb5-config --libs 2> /dev/null"
+                );
+                if (krb5ConfigCheck2.getExitValue() != 0) {
+                    isNative = false;
+                }
+            }
         }
-        MECH = temp;
-    }
 
-    public static void main(String[] argv) throws Exception {
-        GSSManager mgr = GSSManager.getInstance();
-
-        boolean result = true;
-        // Create GSSName and check their equals(), hashCode() impl
-        GSSName name1 = mgr.createName(NAME_STR1,
-            GSSName.NT_HOSTBASED_SERVICE, MECH);
-        GSSName name2 = mgr.createName(NAME_STR2,
-            GSSName.NT_HOSTBASED_SERVICE, MECH);
-        GSSName name3 = mgr.createName(NAME_STR1,
-            GSSName.NT_HOSTBASED_SERVICE, MECH);
-
-        if (!name1.equals(name1) || !name1.equals(name3) ||
-            !name1.equals((Object) name1) ||
-            !name1.equals((Object) name3)) {
-            System.out.println("Error: should be the same name");
-            result = false;
-        } else if (name1.hashCode() != name3.hashCode()) {
-            System.out.println("Error: should have same hash");
-            result = false;
+        if (isNative) {
+
+            final var testCommand = new String[]{"-Dsun.security.jgss.native=true",
+                    Krb5NameEquals.Krb5NameEqualsTest.class.getName()
+            };
+
+            final var result = ProcessTools.executeTestJava(testCommand);
+
+            if (result.getExitValue() != 0) {
+                if (os.equals("Linux") &&
+                    (arch.equals("x86_64") || arch.equals("amd64"))) {
+                    final var javaPropCommand = new String[]{System.getProperty("java.home"),
+                            "-XshowSettings:properties",
+                            "-version 2"
+                    };
+
+                    final var javaPropCommandResult = ProcessTools.executeTestJava(javaPropCommand);
+
+                    final var installationIssue = Arrays.stream(javaPropCommandResult.getStdout().split("\n"))
+                            .anyMatch(line -> line.contains("sun.arch.data.model") && line.contains("32"));
+
```

Comment:
Since this is already a Java program, maybe we can simply `System.getProperty("sun.arch.data.model")` here?

Reply:
In fact, I'm not sure if this is still worth checking. Do we still have 32-bit Linux build?